src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -198,7 +240,7 @@ private async Task<bool> HeartbeatAsync(CancellationToken cancellationToken)
                 newDescription = newDescription.With(heartbeatException: heartbeatException);
             }
 
-            OnDescriptionChanged(newDescription);
+            SetDescriptionIfChanged(newDescription);
```

Comment:
OnDescriptionChanged has been split into three methods:

1. SetDescriptionIfChanged which calls SetDescription if the description changed
2. SetDescription which sets the description and calls OnDescriptionChanged
3. OnDescriptionChanged which now ONLY handles raising the event

Calling SetDescriptionIfChanged executes EXACTLY the same statements as OnDescriptionChanged used to, just that it's now refactored into three smaller methods.